floors/rule.go

```diff
@@ -0,0 +1,504 @@
+	}
+
+	if formatCount > 1 {
+		return catchAll
+	}
+	return value
+}
+
+func getSizeValue(imp openrtb2.Imp) string {
+	size := catchAll
+	width := int64(0)
+	height := int64(0)
+
+	if imp.Banner != nil {
+		width, height = getBannerSize(imp)
+	} else if imp.Video != nil {
+		width = imp.Video.W
+		height = imp.Video.H
+	}
+
+	if width != 0 && height != 0 {
+		size = fmt.Sprintf("%dx%d", width, height)
+	}
+	return size
+}
+
+func getBannerSize(imp openrtb2.Imp) (int64, int64) {
+	width := int64(0)
+	height := int64(0)
+
+	if len(imp.Banner.Format) == 1 {
+		return imp.Banner.Format[0].W, imp.Banner.Format[0].H
+	} else if len(imp.Banner.Format) > 1 {
+		return width, height
+	} else if imp.Banner.W != nil && imp.Banner.H != nil {
+		width = *imp.Banner.W
+		height = *imp.Banner.H
+	}
+	return width, height
+}
+func getDomain(request *openrtb2.BidRequest) string {
+	var value string
+	if request.Site != nil {
+		if len(request.Site.Domain) > 0 {
+			value = request.Site.Domain
+		} else if request.Site.Publisher != nil && len(request.Site.Publisher.Domain) > 0 {
+			value = request.Site.Publisher.Domain
+		}
+	} else if request.App != nil {
+		if len(request.App.Domain) > 0 {
+			value = request.App.Domain
+		} else if request.App.Publisher != nil && len(request.App.Publisher.Domain) > 0 {
+			value = request.App.Publisher.Domain
+		}
+	}
+	return value
+}
+
+func getSiteDomain(request *openrtb2.BidRequest) string {
+	var value string
+	if request.Site != nil {
+		value = request.Site.Domain
+	} else {
+		value = request.App.Domain
+	}
+	return value
+}
+
+func getPublisherDomain(request *openrtb2.BidRequest) string {
+	value := catchAll
+	if request.Site != nil && request.Site.Publisher != nil && len(request.Site.Publisher.Domain) > 0 {
+		value = request.Site.Publisher.Domain
+	} else if request.App != nil && request.App.Publisher != nil && len(request.App.Publisher.Domain) > 0 {
+		value = request.App.Publisher.Domain
+	}
+	return value
+}
+
+func getBundle(request *openrtb2.BidRequest) string {
+	value := catchAll
+	if request.App != nil && len(request.App.Bundle) > 0 {
+		value = request.App.Bundle
+	}
+	return value
+}
+
+func getgptslot(imp openrtb2.Imp) string {
+	value := catchAll
+	adsname, err := jsonparser.GetString(imp.Ext, "data", "adserver", "name")
+	if err == nil && adsname == "gam" {
+		gptSlot, _ := jsonparser.GetString(imp.Ext, "data", "adserver", "adslot")
+		if gptSlot != "" {
+			value = gptSlot
+		}
+	} else {
+		value = getpbadslot(imp)
+	}
+	return value
+}
+
+func extractChanelNameFromBidRequestExt(bidRequest *openrtb2.BidRequest) string {
+	requestExt := &openrtb_ext.ExtRequest{}
+	if bidRequest == nil {
+		return catchAll
+	}
+
+	if len(bidRequest.Ext) > 0 {
+		err := json.Unmarshal(bidRequest.Ext, &requestExt)
+		if err != nil {
+			return catchAll
+		}
+	}
+
+	if requestExt.Prebid.Channel != nil {
+		return requestExt.Prebid.Channel.Name
+	}
+	return catchAll
+}
+
+func getpbadslot(imp openrtb2.Imp) string {
```

Comment:
Can this be renamed for camel-case `getpbadslot` --> `getPbAdSlot`

Reply:
renamed to `getPbAdSlot`